src/realm/group_writer.cpp

```diff
@@ -887,10 +986,11 @@ void GroupWriter::FreeList::merge_adjacent_entries_in_freelist()
     }
 }
 
-void GroupWriter::FreeList::move_free_in_file_to_size_map(std::multimap<size_t, size_t>& size_map)
+void GroupWriter::move_free_in_file_to_size_map(const std::vector<GroupWriter::FreeSpaceEntry>& list,
```

Comment:
This is a tricky bit of code, is there a way to independently test some of the edge cases you added here apart from the general integration tests?

Reply:
I like that idea. I don't think this particular file is the biggest risk here. Have you seen the `backdate` function? We should explore that possibility more.

Reply:
The backdate function is awesome. Just saying.